src/lib/auth.ts

```diff
@@ -0,0 +1,44 @@
+export interface CustomerSignInResult {
+  access_token: string;
+  expires_in: number;
+  scope: string;
+  token_type: string;
+  refresh_token?: string;
+}
+
+const projectKey = process.env.NEXT_PUBLIC_PROJECT_KEY;
+const authUrl = process.env.NEXT_PUBLIC_AUTH_URL;
+const clientId = process.env.NEXT_PUBLIC_CLIENT_ID;
+const clientSecret = process.env.NEXT_PUBLIC_CLIENT_SECRET;
+const scopes = process.env.NEXT_PUBLIC_SCOPES ?? '';
+
+const getBasicAuthHeader = (): string =>
+  'Basic ' + Buffer.from(`${clientId}:${clientSecret}`).toString('base64');
```

Comment:
const smth = `${clientId}:${clientSecret}`;
`Basic ${Buffer.from(smth).toString('base64')}`
Do not combine template strings with concatenation, looks sloppy.